lib/dict.c

```diff
@@ -86,11 +75,11 @@ int rc_read_dictionary (rc_handle *rh, char const *filename)
 			optstr[0] = '\0';
 			/* Read the ATTRIBUTE line */
 			if (sscanf (buffer, "%63s%63s%63s%63s%63s", dummystr, namestr,
-				    valstr, typestr, optstr) < 4)
+				valstr, typestr, optstr) < 4)
 			{
-				rc_log(LOG_ERR, "rc_read_dictionary: invalid attribute on line %d of dictionary %s",
-					 line_no, filename);
-				fclose(dictfd);
+				rc_log(LOG_ERR, 
+					"rc_dict_init: invalid attribute on line %d of "
```

Comment:
hmm, since you kept the filename as parameter, why remove it from here? Wouldn't it be more natural if it was kept here so we don't regress in debugging? An approach could be to use another variable for printing, something like:
```
const char *pfilename = filename;
if (pfilename == NULL) pfilename = "memory";

/* and use pfilename later for printing */
...
```